### ISSUE TITLE
BaseConfig does not work recursively  in Dict

### ISSUE THREAD

**Issue by rkrell** [mkaranasou/pyaml_env#1]
Hi, great small library.

In version 1.0.0., I found one issue: BaseConfig does not work as documented for me. It does not recurse deeper into parsed dictionaries.

Example:

config.yaml:
```yaml
db:
  connection: mysql+pymysql://user1:pass1@localhost/db_name?charset=utf8
```

```python
cfg = BaseConfig(parse_config('config.yaml))
```
returns an object of type `BaseConfig` with an attribute `db` of type dictionary, thus

```python
cfg.db.connection
```
is not accessible in that way, the only way to get the connection is:

```python
cfg.db['connection']
```

**Comment by mkaranasou** [comment]
Hi, thanks for pointing it out, it was a bit premature to release the BaseConfig to be honest, my apologies for the confusion. I just merged a fix and the relevant tests for this, v1.0.1 should be up soon in pypi  :) 
Let me know if it is ok for you or if you have any other issues.

**Comment by rkrell** [comment]
> Hi, thanks for pointing it out, it was a bit premature to release the BaseConfig to be honest, my apologies for the confusion. I just merged a fix and the relevant tests for this, v1.0.1 should be up soon in pypi :)
> Let me know if it is ok for you or if you have any other issues.

Many thanks for this super-fast response. I'll check early.

**Comment by rkrell** [comment]
This problem has been fixed in 1.0.1.

Thank you.
